tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +781,295 @@ int main(int argc, char **argv)
             }
         }
 
+        /* Test ChaCha20 boosting behaviour */
+        {
+            EXPECT_SUCCESS(s2n_enable_tls13_in_test());
+
+            /* The server chose a (malformed) security policy where ChaCha20 boosting is toggled on but 
+             * the cipher preferences does not contain ChaCha20. In this case, ChaCha20 boosting behaviour is off.
+             */
+            {
+                static struct s2n_cipher_suite *cipher_suites_without_chacha20[] = {
+                    &s2n_tls13_aes_256_gcm_sha384,                  /* 0x13,0x02 */
+                    &s2n_tls13_aes_128_gcm_sha256,                  /* 0x13,0x01 */
+                };
+
+                const struct s2n_cipher_preferences cipher_preference_without_chacha20_and_boosting = {
+                    .count = s2n_array_len(cipher_suites_without_chacha20),
+                    .suites = cipher_suites_without_chacha20,
+                    .allow_chacha20_boosting = true
+                };
+
+                const struct s2n_security_policy test_no_chacha20_and_boosted = {
+                    .minimum_protocol_version = S2N_SSLv3,
+                    .cipher_preferences = &cipher_preference_without_chacha20_and_boosting,
+                    .kem_preferences = &kem_preferences_null,
+                    .signature_preferences = &s2n_signature_preferences_20201021,
+                    .ecc_preferences = &s2n_ecc_preferences_test_all,
+                };
```

Comment:
This will be the same every time. I'd really suggest only building it once. You can change the cipher preferences either by changing the pointer on the security policy or by changing the value of the cipher preferences it points to without using its const reference.

Like:
```
// test suite context
{
    struct s2n_security_policy s2n_security_policy = {
        .minimum_protocol_version = S2N_SSLv3,
        .cipher_preferences = &cipher_preferences,
        .kem_preferences = &kem_preferences_null,
        .signature_preferences = &s2n_signature_preferences_20201021,
        .ecc_preferences = &s2n_ecc_preferences_test_all,
    };
    
    DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
    EXPECT_NOT_NULL(conn);
    conn->security_policy_override = &s2n_security_policy;

   // specific test context
   {
       struct s2n_cipher_suite *cipher_suites[] = {
             &s2n_tls13_aes_256_gcm_sha384,
             &s2n_tls13_aes_128_gcm_sha256, 
       };
       const struct cipher_preferences cipher_preferences = { 
           .count = s2n_array_len(cipher_suites),
           .suites = cipher_suites,
           .allow_chacha20_boosting = true
      }
      security_policy->cipher_preferences = &cipher_preferences;

      // rest of test
   }
}
```
Or:
```
// test suite context
{
    struct s2n_cipher_preferences cipher_preferences = { 0 };
    const struct s2n_security_policy s2n_security_policy = {
        .minimum_protocol_version = S2N_SSLv3,
        .cipher_preferences = &cipher_preferences,
        .kem_preferences = &kem_preferences_null,
        .signature_preferences = &s2n_signature_preferences_20201021,
        .ecc_preferences = &s2n_ecc_preferences_test_all,
    };
    
    DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
    EXPECT_NOT_NULL(conn);
    conn->security_policy_override = &s2n_security_policy;

   // specific test context
   {
       struct s2n_cipher_suite *cipher_suites[] = {
             &s2n_tls13_aes_256_gcm_sha384,
             &s2n_tls13_aes_128_gcm_sha256, 
       };
       cipher_preferences = (struct s2n_cipher_preferences){ 
           .count = s2n_array_len(cipher_suites),
           .suites = cipher_suites,
           .allow_chacha20_boosting = true
      }

      // rest of test
   }
}
```